src/apps/beefy/positions.ts

```diff
@@ -88,38 +184,34 @@ const hook: PositionsHook = {
 
     const prices = await getBeefyLpsPrices()
 
-    return userVaults.map(
-      (vault): AppTokenPositionDefinition => ({
-        type: 'app-token-definition',
-        networkId,
-        address: vault.earnedTokenAddress.toLowerCase(),
-        tokens: [
-          {
-            address: vault.tokenAddress.toLowerCase(),
-            networkId,
-            fallbackPriceUsd: prices[vault.id]
-              ? toSerializedDecimalNumber(prices[vault.id])
-              : undefined,
-          },
-        ],
-        displayProps: () => {
-          return {
-            title: vault.name + (vault.status === 'eol' ? ' (Retired)' : ''),
-            description: 'Vault',
-            imageUrl:
-              'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/beefy.png',
-          }
-        },
-        pricePerShare: async ({ tokensByTokenId }) => {
-          const tokenId = getTokenId({
-            address: vault.tokenAddress,
-            networkId,
-          })
-          const { decimals } = tokensByTokenId[tokenId]
-          return [toDecimalNumber(BigInt(vault.pricePerFullShare), decimals)]
-        },
-      }),
+    const clmVaults = userVaults.filter(
+      (vault) => vault.strategyTypeId === 'cowcentrated',
     )
+    const info =
+      clmVaults.length === 0
+        ? []
+        : ((await client.readContract({
+            code: beefyClmVaultsMulticallBytecode,
+            abi: beefyClmVaultsMulticallAbi,
+            functionName: 'getUserVaults',
+            args: [address, clmVaults.map((vault) => vault.earnedTokenAddress)],
+          })) as ClmVaultBalanceInfo[])
```

Comment:
nit: is it possible to let TS infer the return type from the ABI?

Reply:
I spent like 20 minutes the other day trying to do it and I couldn't figure it out, and today I realized I was missing the `as const` in the abi in under a minute 😂